org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/WrapperValidatorTest.java

```diff
@@ -47,21 +49,26 @@ public void setProperty() throws Exception {
 	}
 
 	@After
-	public void clearProperty() {
+	public void clearProperty() throws IOException {
 		System.clearProperty("gradle.checksum.cacheDir");
+		WrapperValidator.clear();
 	}
 
 	@Test
 	public void testGradleWrapper() throws Exception {
 		File file = new File(getSourceProjectDirectory(), "gradle/simple-gradle");
 		assertTrue(file.isDirectory());
+		File sha256Directory = WrapperValidator.getSha256CacheFile();
+		assertTrue(sha256Directory.isDirectory());
 		ValidationResult result = new WrapperValidator().checkWrapper(file.getAbsolutePath());
 		assertTrue(result.isValid());
-		File sha256Directory = WrapperValidator.getSha256CacheFile();
 		// test cache
-		file = new File(sha256Directory, "gradle-6.4-wrapper.jar.sha256");
+		assertTrue(sha256Directory.isDirectory());
+		String message = Files.list(Paths.get(sha256Directory.getAbsolutePath())).collect(Collectors.toList()).toString();
+		file = new File(sha256Directory, "gradle-6.3-wrapper.jar.sha256");
+		assertTrue(message, file.isFile());
```

Comment:
I saw this fail if WrapperValidator.clear() is not set in setProperty()

Reply:
checking...